src/meta/src/rpc/server.rs

```diff
@@ -171,316 +153,312 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
-    sub_tasks.push((election_handle, election_shutdown));
-    if cfg!(not(test)) {
-        sub_tasks.push(
-            ClusterManager::start_heartbeat_checker(cluster_manager, Duration::from_secs(1)).await,
+    #[tokio::test]
+    async fn test_failover_1() {
+        let leader_count = _test_failover(1, 9012, 1012).await;
+        assert_eq!(
+            leader_count, 0,
+            "Expected to have 1 leader, instead got {} leaders",
+            leader_count
         );
-        sub_tasks.push(GlobalBarrierManager::start(barrier_manager).await);
     }
-    let (idle_send, idle_recv) = tokio::sync::oneshot::channel();
-    sub_tasks.push(
-        IdleManager::start_idle_checker(env.idle_manager_ref(), Duration::from_secs(30), idle_send)
-            .await,
-    );
-    let shutdown_all = async move {
-        for (join_handle, shutdown_sender) in sub_tasks {
-            if let Err(_err) = shutdown_sender.send(()) {
-                // Maybe it is already shut down
-                continue;
-            }
-            if let Err(err) = join_handle.await {
-                tracing::warn!("Failed to join shutdown: {:?}", err);
-            }
-        }
-    };
 
-    let (svc_shutdown_tx, svc_shutdown_rx) = tokio::sync::oneshot::channel::<()>();
+    #[tokio::test]
+    async fn test_failover_3() {
+        let leader_count = _test_failover(3, 1100, 1200).await;
+        assert_eq!(
+            leader_count, 1,
+            "Expected to have 1 leader, instead got {} leaders",
+            leader_count
+        );
+    }
 
-    // Start services.
-    tracing::info!("Starting leader services");
-    let join_handle = tokio::spawn(async move {
-        tonic::transport::Server::builder()
-            .layer(MetricsMiddlewareLayer::new(meta_metrics.clone()))
-            .add_service(HeartbeatServiceServer::new(heartbeat_srv))
-            .add_service(ClusterServiceServer::new(cluster_srv))
-            .add_service(StreamManagerServiceServer::new(stream_srv))
-            .add_service(HummockManagerServiceServer::new(hummock_srv))
-            .add_service(NotificationServiceServer::new(notification_srv))
-            .add_service(DdlServiceServer::new(ddl_srv))
-            .add_service(UserServiceServer::new(user_srv))
-            .add_service(ScaleServiceServer::new(scale_srv))
-            .add_service(HealthServer::new(health_srv))
-            .add_service(BackupServiceServer::new(backup_srv))
-            .serve_with_shutdown(address_info.listen_addr, async move {
-                tokio::select! {
-                    _ = tokio::signal::ctrl_c() => {},
-                    _ = svc_shutdown_rx => {
-                        shutdown_all.await;
-                    },
-                    _ = idle_recv => {
-                        shutdown_all.await;
-                    },
-                }
-            })
-            .await
-            .unwrap();
-    });
+    #[tokio::test]
+    async fn test_failover_10() {
```

Comment:
Ditto, not necessary. `test_failover_1` and `test_failover_3` is already enough.

Reply:
Alright. I will only keep 1 and 3 here.